In [0]:

import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import accuracy_score
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import accuracy_score
from sklearn.grid_search import GridSearchCV
from sklearn import datasets, svm
import matplotlib.pyplot as plt
from sklearn.model_selection import LeaveOneOut
#df = pd.read_csv('wml_data.csv', header = 0)
#df = pd.read_csv('wml_avn_fan_op.csv', header = 0)

C:\Users\Public\lib\site-packages\sklearn\cross_validation.py:44: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)
C:\Users\Public\lib\site-packages\sklearn\grid_search.py:43: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. This module will be removed in 0.20.
  DeprecationWarning)


In [0]:
#multivariate linear regression NEW BLOCK 
#80/20 split- 20% training data

#train, test = train_test_split(fifa_dataset, test_size=0.2)  // fifa_dataset here is a panda dataframe



data_avn = pd.read_csv('wml_avn_label.csv', header = 0)

data_fan = pd.read_csv('wml_fan_label.csv', header = 0)

data_avn_fan = pd.read_csv('wml_avn_fan_label.csv', header = 0)

data_avn_fan_op = pd.read_csv('wml_avn_fan_op_label.csv', header = 0)

data_op = pd.read_csv('wml_op_label.csv', header = 0)
#label = pd.read_csv('wml_label.csv', header = 0)





# for avn_fan_op
df2 = data_avn_fan_op  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

#x_train = train.iloc[:,1:22] #- Except OP mean median std----------------------------------------------
x_train = train.iloc[:,1:25]
x_test = test.iloc[:,1: 25]

print( x_train , y_train_true)



     0       0.1  0.2      0.3       0.4  0.5      0.6   1  0.7  1.1  ...   \
79   2  0.416670    0  0.79296  0.916670  0.0  1.56430   3    4    1  ...    
110  2  0.333330    0  0.65134  0.166670  0.0  0.57735   4    2    4  ...    
95   0  0.000000    0  0.00000  0.500000  0.0  0.79772   1    2    6  ...    
41   2  0.166670    0  0.57735  0.583330  0.0  0.90034   3    2    2  ...    
28   1  0.083333    0  0.28868  0.833330  0.0  1.02990   1    2    1  ...    
58   0  0.000000    0  0.00000  0.333330  0.0  0.77850   1    2    2  ...    
4    0  0.000000    0  0.00000  0.166670  0.0  0.57735   1    2   11  ...    
82   2  0.166670    0  0.57735  0.750000  0.0  1.28810   3    4    5  ...    
22   1  0.083333    0  0.28868  0.166670  0.0  0.57735   4    2    9  ...    
124  0  0.000000    0  0.00000  0.000000  0.0  0.00000   1    0    1  ...    
83   3  0.916670    0  1.16450  0.166670  0.0  0.57735   4    2    9  ...    
85   1  0.083333    0  0.28868  0.666670  0.0  1.23090  12    4 

In [0]:
# for avn_fan
df2 = data_avn_fan  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

x_train = train.iloc[:,1:17] #-----------------------------------------------

x_test = test.iloc[:,1: 17]

print( x_train , y_train_true)

     0       0.1  0.2      0.3       0.4  0.5      0.6   1  0.7  1.1  0.8  \
18   2  0.250000    0  0.62158  0.166670  0.0  0.38925   9    1    4    0   
22   1  0.083333    0  0.28868  0.166670  0.0  0.57735   4    2    9    1   
71   0  0.000000    0  0.00000  0.416670  0.0  0.79296   1    2    1    0   
76   2  0.333330    0  0.77850  1.083300  0.0  1.78160   3    4    1    4   
44   1  0.083333    0  0.28868  0.000000  0.0  0.00000   3    0    1    1   
77   2  0.166670    0  0.57735  1.666700  2.0  1.66970  10    4    3    0   
21   1  0.083333    0  0.28868  0.166670  0.0  0.57735   3    2    3    1   
95   0  0.000000    0  0.00000  0.500000  0.0  0.79772   1    2    6    0   
38   2  0.250000    0  0.62158  0.416670  0.0  0.79296   4    2    4    3   
78   1  0.083333    0  0.28868  0.916670  0.0  1.56430   5    4    3    1   
29   0  0.000000    0  0.00000  0.000000  0.0  0.00000   1    0    1    0   
53   0  0.000000    0  0.00000  0.166670  0.0  0.57735   1    2    9    0   

In [0]:
# for avn
df2 = data_avn  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

x_train = train.iloc[:,1:8] #-----------------------------------------------

x_test = test.iloc[:,1: 8]

print( x_train , y_train_true)

     0   1  0.1  0.2
127  0   1    0    0
93   2   4    2    0
124  0   1    0    0
0    0   1    0    0
73   4   4    7    0
57   2   1    4    4
1    0   1    0    0
37   0   1    0    0
96   1   4    1    0
90   4   4    5    0
125  0   1    0    0
83   3   4    3    8
19   4   8    0   12
65   2   8    2    2
101  0   1    0    0
50   2   8    0    2
106  0   1    0    0
69   0   1    0    0
17   2   5    3    2
97   3   4    5    0
100  2  10    0    2
112  0   1    0    0
78   1   5    1    0
109  1   3    2    0
80   2   4    4    5
16   0   1    0    0
9    2   9    0    2
79   2   3    5    0
31   0   1    0    0
4    0   1    0    0
..  ..  ..  ...  ...
8    1   5    1    0
95   0   1    0    0
21   1   3    1    0
39   0   1    0    0
77   2  10    0    2
89   4   9    2    8
122  1   6    1    0
42   0   1    0    0
32   1   3    2    0
74   4   9    4   12
28   1   1    1    0
20   1   8    0    2
35   2   2    3    0
82   2   3    2    0
75   2   3    6    0
115  1   4   

In [0]:
# for fan
df2 = data_fan  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

x_train = train.iloc[:,1:8] #-----------------------------------------------

x_test = test.iloc[:,1: 8]

print( x_train , y_train_true)

In [0]:
# for op
df2 = data_op  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

x_train = train.iloc[:,1:8] #- with OP extra3----------------------------------------------

x_test = test.iloc[:,1: 8]

print( x_train , y_train_true)

     0   1  0.1  0.2
22   0   1    0    0
5    1   5    1    0
98   1  10    0    1
12   0   1    0    0
117  2   7    1    2
37   2   1    2    4
90   2   5    2    0
26   2   8    0    2
69   2   8    0    2
11   0   1    0    0
120  0   1    0    0
49   2   1    2    0
122  0   1    0    0
111  0   1    0    0
109  0   1    0    0
126  0   1    0    0
51   1   8    0    1
78   4   2    8    4
9    1   3    1    0
67   2   2    3    0
40   4   8    4    6
94   1   8    0    1
101  2   8    0    4
21   2   2    2    0
46   0   1    0    0
24   4  10    1    4
60   0   1    0    0
13   0   1    0    0
131  0   1    0    0
119  2   7    0    4
..  ..  ..  ...  ...
81   2   3    2    2
33   2   7    0    3
4    0   1    0    0
100  0   1    0    0
87   0   1    0    0
47   0   1    0    0
23   0   1    0    0
110  2   3    2    0
6    0   1    0    0
105  1   9    0    1
74   2   5    2    4
79   4   1    4    0
43   0   1    0    0
116  1   9    0    1
121  2   8    1    2
18   1   3   

# GRID SEARCH hyperparameter tuning

# GENERAL MODEL fit & Training cell

In [0]:
'''
x_train =  #-----------------------------------------------

x_test = 

y_train_true = 
y_test_pred
y_test_true =  #----------------------------------------
'''
model.fit(x_train, y_train_true.values.ravel()) ###########################################

#mean absolute value for training data
##data = train[target]

 
 
##predict =  model.predict(train[features])
y_train_pred =  model.predict(x_train)  #--------------------------------------------------



print (y_train_true.values.ravel(),y_train_pred)

y_test_pred = model.predict(x_test)

print(y_test_true.values.ravel(), y_test_pred)

[2 2 1 2 1 2 1 3 2 3 3 1 2 2 1 3 1 1 1 1 3 1 3 3 1 3 3 1 3 3 2 3 3 3 3 2 1
 2 3 1 3 3 3 1 3 3 2 3 1 1 3 2 3 3 3 1 2 1 1 3 2 2 3 2 3 2 3 2 1 2 3 3 2 2
 1 2 3 2 3 1 3 1 2 1 3 3 1 1 2 2 3 1 1 1 3 2 2 1 1 1 3 2 1 1 1 2 3 3 1 1 2
 3 2 1 1 2 1 1] [3 2 3 2 1 2 3 3 2 3 3 2 1 3 1 1 1 3 1 1 1 1 3 3 1 3 3 3 1 3 1 1 3 3 3 1 1
 1 3 1 3 3 3 1 1 3 3 3 1 1 1 1 2 3 3 1 1 1 1 3 2 2 3 2 3 3 3 1 1 2 1 2 1 1
 1 3 3 1 3 2 3 3 3 1 2 3 1 1 1 1 3 1 1 1 3 1 3 3 2 2 3 3 1 1 3 3 3 3 1 1 1
 2 2 1 1 3 1 1]
[1 1 3 2 2 2 1 1 1 1 3 1 2 3] [3 2 2 1 2 3 2 1 1 2 3 1 1 1]


In [0]:
accuracy = r2_score(y_test_true, y_test_pred)

NameError: name 'r2_score' is not defined

# accuracy cell 

In [0]:
from sklearn import metrics
print("Accuracy:",metrics.accuracy_score(y_test_true, y_test_pred))

print("Train Accuracy:",metrics.accuracy_score(y_train_true, y_train_pred))


from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

results = confusion_matrix(y_test_true, y_test_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_test_true, y_test_pred) ) 
print ('Report : ')
print (classification_report(y_test_true, y_test_pred) )

Accuracy: 0.357142857143
Train Accuracy: 0.610169491525
Confusion Matrix :
[[3 3 1]
 [2 1 1]
 [1 1 1]]
Accuracy Score : 0.357142857143
Report : 
             precision    recall  f1-score   support

          1       0.50      0.43      0.46         7
          2       0.20      0.25      0.22         4
          3       0.33      0.33      0.33         3

avg / total       0.38      0.36      0.37        14



# K-fold cross validation training & score

In [0]:
from sklearn.model_selection import cross_val_score
#scores = cross_val_score(model , X = x_train , y = y_train_true , cv = x_train.shape[0]) 
scores = cross_val_score(model , X = x_train , y = y_train_true.values.ravel() , cv = 30) 
print (scores)

[ 0.33333333  0.          0.5         0.6         0.4         0.4         0.8
  0.6         0.6         0.4         0.8         0.2         1.          0.
  0.33333333  1.          0.33333333  0.          0.          0.33333333
  0.33333333  0.66666667  0.33333333  0.33333333  0.66666667  0.66666667
  0.33333333  0.33333333  1.          0.33333333]


data_avn = pd.read_csv('wml_avn_label.csv', header = 0)

data_fan = pd.read_csv('wml_fan_label.csv', header = 0)

data_avn_fan = pd.read_csv('wml_avn_fan_label.csv', header = 0)

data_avn_fan_op = pd.read_csv('wml_avn_fan_op_label.csv', header = 0)

data_op 

# LEAVE ONE OUT USING SVM (named as model) 100%  data ONLY AVN & FAN

In [0]:
'''
parameter_candidates = [
  {'C': [1, 10, 100, 1000], 'kernel': ['linear']},
  {'C': [1, 10, 100, 1000], 'gamma': [0.001, 0.0001], 'kernel': ['rbf']},
]
# Create a classifier object with the classifier and parameter candidates
model = GridSearchCV(estimator=svm.SVC(), param_grid=parameter_candidates, n_jobs=-1)
'''
parameter_candidates = [
  {'C': [ 100], 'kernel': ['linear'] }

]
# Create a classifier object with the classifier and parameter candidates
model = GridSearchCV(estimator=svm.SVC(), param_grid=parameter_candidates, n_jobs=-1)


In [0]:


from sklearn.model_selection import LeaveOneOut ########LEAVE ONE OUT
import numpy as np
#x = np.array([[1, 2], [3, 4]])
x_train = data_avn_fan.iloc[:,1:17] #----------------
y_train = data_avn_fan.iloc[:,[0]]


x = np.array(x_train)
y = np.array(y_train.values.ravel())

############# Standard Scalar
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
#scaler.fit_transform(x.values.reshape(-1, 1))
x = scaler.fit_transform(x)
#X_test = scaler.transform(X_test)

loo = LeaveOneOut()
loo.get_n_splits(x)
# x_train mane whole data nea dorkar ekhane
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        #print (y_train_true.values.ravel(),y_train_pred)

        #y_test_pred = model.predict(x_test)        
#model.fit(x_train, y_train_true.values.ravel()) ###########################################

print('Best C:',model.best_estimator_.C) 
print('Best Kernel:',model.best_estimator_.kernel)
print('Best Gamma:',model.best_estimator_.gamma)

from sklearn import metrics
from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
print (classification_report(y_train, y_train_pred) )

Best C: 100
Best Kernel: linear
Best Gamma: auto
Train Accuracy: 0.740458015267
Confusion Matrix :
[[31 18]
 [16 66]]
Accuracy Score : 0.740458015267
Report : 
             precision    recall  f1-score   support

          1       0.66      0.63      0.65        49
          2       0.79      0.80      0.80        82

avg / total       0.74      0.74      0.74       131



# SVM model(GRID search scaler) AVN, FAN_class 2
Best C: 100
Best Kernel: linear
Best Gamma: auto
Train Accuracy: 0.740458015267
Confusion Matrix :
[[31 18]
 [16 66]]
Accuracy Score : 0.740458015267
Report : 
             precision    recall  f1-score   support

          1       0.66      0.63      0.65        49
          2       0.79      0.80      0.80        82

avg / total       0.74      0.74      0.74       131
# SVM model AVN, FAN, OP
Train Accuracy: 0.770992366412
Confusion Matrix :
[[36 13]
 [17 65]]
Accuracy Score : 0.770992366412
Report : 
             precision    recall  f1-score   support

          1       0.68      0.73      0.71        49
          2       0.83      0.79      0.81        82

avg / total       0.78      0.77      0.77       131
# SVM model AVN
Train Accuracy: 0.679389312977
Confusion Matrix :
[[31 18]
 [24 58]]
Accuracy Score : 0.679389312977
Report : 
             precision    recall  f1-score   support

          1       0.56      0.63      0.60        49
          2       0.76      0.71      0.73        82

avg / total       0.69      0.68      0.68       131
# SVM model  FAN
Train Accuracy: 0.63358778626
Confusion Matrix :
[[ 2 47]
 [ 1 81]]
Accuracy Score : 0.63358778626
Report : 
             precision    recall  f1-score   support

          1       0.67      0.04      0.08        49
          2       0.63      0.99      0.77        82

avg / total       0.65      0.63      0.51       131
# SVM model  OP
Train Accuracy: 0.625954198473
Confusion Matrix :
[[ 0 49]
 [ 0 82]]
Accuracy Score : 0.625954198473
Report : 
             precision    recall  f1-score   support

          1       0.00      0.00      0.00        49
          2       0.63      1.00      0.77        82

avg / total       0.39      0.63      0.48       131

# SVM model AVN, FAN, OP 

In [0]:
'''
parameter_candidates = [
  {'C': [1, 10, 100, 1000], 'kernel': ['linear']},
  {'C': [1, 10, 100, 1000], 'gamma': [0.001, 0.0001], 'kernel': ['rbf']},
]
# Create a classifier object with the classifier and parameter candidates
model = GridSearchCV(estimator=svm.SVC(), param_grid=parameter_candidates, n_jobs=-1)



from sklearn.model_selection import LeaveOneOut ########LEAVE ONE OUT
import numpy as np

'''
#x = np.array([[1, 2], [3, 4]])
x_train = data_avn_fan_op.iloc[:,1:25] #----------------
y_train = data_avn_fan_op.iloc[:,[0]]

from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
x = np.array(x_train)
y = np.array(y_train.values.ravel())

x = scaler.fit_transform(x)

loo = LeaveOneOut()
loo.get_n_splits(x)
# x_train mane whole data nea dorkar ekhane
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        
from sklearn import metrics

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))


from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
print (classification_report(y_train, y_train_pred) )

NameError: name 'data_avn_fan_op' is not defined

# SVM & AVN

In [0]:
x_train = data_avn.iloc[:,1:8] #----------------
y_train = data_avn.iloc[:,[0]]


x = np.array(x_train)
y = np.array(y_train.values.ravel())

x = scaler.fit_transform(x)

loo = LeaveOneOut()
loo.get_n_splits(x)
# x_train mane whole data nea dorkar ekhane
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
print (classification_report(y_train, y_train_pred) )

Train Accuracy: 0.679389312977
Confusion Matrix :
[[31 18]
 [24 58]]
Accuracy Score : 0.679389312977
Report : 
             precision    recall  f1-score   support

          1       0.56      0.63      0.60        49
          2       0.76      0.71      0.73        82

avg / total       0.69      0.68      0.68       131



# SVM_ & FAN  OP combinations 

In [0]:
#x_train = data_op.iloc[:,1:8] #----------------
#y_train = data_op.iloc[:,[0]]
import numpy as np
data_avn_op = pd.read_csv('op_avn_label.csv', header = 0)
data_fan_op = pd.read_csv('op_fan_label.csv', header = 0)
x_train = data_fan_op.iloc[:,1:16] #----------------
y_train = data_fan_op.iloc[:,[0]]

x = np.array(x_train)
y = np.array(y_train.values.ravel())
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
x = scaler.fit_transform(x)

loo = LeaveOneOut()
loo.get_n_splits(x)
# x_train mane whole data nea dorkar ekhane
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        
from sklearn import metrics

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))


from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
y_train_pred =  model.predict(x_train)
print (classification_report(y_train, y_train_pred) )

Train Accuracy: 0.664122137405
Confusion Matrix :
[[24 25]
 [19 63]]
Accuracy Score : 0.664122137405
Report : 
             precision    recall  f1-score   support

          1       0.56      0.49      0.52        49
          2       0.72      0.77      0.74        82

avg / total       0.66      0.66      0.66       131



op_Avn
Train Accuracy: 0.679389312977
Confusion Matrix :
[[27 22]
 [20 62]]
Accuracy Score : 0.679389312977
Report : 
             precision    recall  f1-score   support

          1       0.57      0.55      0.56        49
          2       0.74      0.76      0.75        82

avg / total       0.68      0.68      0.68       131
OP_FAN
Accuracy: 0.664122137405
Confusion Matrix :
[[24 25]
 [19 63]]
Accuracy Score : 0.664122137405
Report : 
             precision    recall  f1-score   support

          1       0.56      0.49      0.52        49
          2       0.72      0.77      0.74        82

avg / total       0.66      0.66      0.66       131

# Random forest ----leave one out & AVN_FAN

In [0]:
#Import Random Forest Model
from sklearn.ensemble import RandomForestClassifier

clf=RandomForestClassifier(n_estimators=400)
model = clf

#clf.fit(x_train,y_train_true)
#y_test_pred=clf.predict(x_test)
#x_train = data_avn_fan.iloc[:,1:17]
#x_train = data_avn.iloc[:,1:8] #----------------
#y_train = data_avn.iloc[:,[0]]
import numpy as np
data_avn_op = pd.read_csv('op_avn_label.csv', header = 0)
data_fan_op = pd.read_csv('op_fan_label.csv', header = 0)
x_train = data_fan_op.iloc[:,1:16] #----------------
y_train = data_fan_op.iloc[:,[0]]

x = np.array(x_train)
y = np.array(y_train.values.ravel())
'''
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
x = scaler.fit_transform(x)
'''

loo = LeaveOneOut()
loo.get_n_splits(x)
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        #print (y_train_true.values.ravel(),y_train_pred)

        #y_test_pred = model.predict(x_test)        
#model.fit(x_train, y_train_true.values.ravel()) ###########################################

from sklearn import metrics
from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 

y_train_pred =  model.predict(x)
print ('Report : ')
#print (classification_report(y_train, y_train_pred) )
print (classification_report(y, y_train_pred) )

Train Accuracy: 0.793893129771
Confusion Matrix :
[[45  4]
 [23 59]]
Accuracy Score : 0.793893129771
Report : 
             precision    recall  f1-score   support

          1       0.65      0.92      0.76        49
          2       0.94      0.71      0.81        83

avg / total       0.83      0.79      0.79       132



# Random Forest model AVN, FAN_class 2
Train Accuracy: 0.885496183206
Confusion Matrix :
[[45  4]
 [11 71]]
Accuracy Score : 0.885496183206
Report : 
             precision    recall  f1-score   support

          1       0.80      0.92      0.86        49
          2       0.95      0.87      0.90        82

avg / total       0.89      0.89      0.89       131
# Random Forest model AVN, FAN, OP
Train Accuracy: 0.916030534351
Confusion Matrix :
[[47  2]
 [ 9 73]]
Accuracy Score : 0.916030534351
Report : 
             precision    recall  f1-score   support

          1       0.84      0.96      0.90        49
          2       0.97      0.89      0.93        82

avg / total       0.92      0.92      0.92       131
# Random Forest model AVN
Train Accuracy: 0.725190839695
Confusion Matrix :
[[39 10]
 [26 56]]
Accuracy Score : 0.725190839695
Report : 
             precision    recall  f1-score   support

          1       0.60      0.80      0.68        49
          2       0.85      0.68      0.76        82

avg / total       0.76      0.73      0.73       131
# Random Forest model  FAN
Train Accuracy: 0.763358778626
Confusion Matrix :
[[42  7]
 [24 58]]
Accuracy Score : 0.763358778626
Report : 
             precision    recall  f1-score   support

          1       0.64      0.86      0.73        49
          2       0.89      0.71      0.79        82

avg / total       0.80      0.76      0.77       131
# Random Forest model  OP
Train Accuracy: 0.75572519084
Confusion Matrix :
[[19 30]
 [ 2 80]]
Accuracy Score : 0.75572519084
Report : 
             precision    recall  f1-score   support

          1       0.90      0.39      0.54        49
          2       0.73      0.98      0.84        83

avg / total       0.79      0.76      0.73       132
# OP_AVN
 Accuracy: 0.854961832061
Confusion Matrix :
[[39 10]
 [ 9 73]]
Accuracy Score : 0.854961832061
Report : 
             precision    recall  f1-score   support

          1       0.80      0.80      0.80        49
          2       0.88      0.88      0.88        83

avg / total       0.85      0.85      0.85       132
# OP_FAN
Accuracy: 0.793893129771
Confusion Matrix :
[[45  4]
 [23 59]]
Accuracy Score : 0.793893129771
Report : 
             precision    recall  f1-score   support

          1       0.65      0.92      0.76        49
          2       0.94      0.71      0.81        83

avg / total       0.83      0.79      0.79       132

# ERROR for REGression

In [0]:



test_data_error = mean_absolute_error(y_test_true, y_test_pred)
accuracy

In [0]:
#define model I am using
model = LinearRegression()
#training process
##model.fit(train[features], train[target])
from sklearn.metrics import f1_score

model.fit(data_avn, target_1col)
#mean absolute value for training data
##data = train[target]
true_target = target_1col
##predict =  model.predict(train[features])
predict =  model.predict(data_avn)


training_error = mean_absolute_error(true_target, predict)
accuracy = r2_score(true_target, predict)
#accuracy = f1_score(true_target, predict, average='micro')
print(predict)
accuracy




[[  2799.4262873 ]
 [  2833.22335692]
 [  2492.58545066]
 [  2551.87960068]
 [  2000.29881734]
 [  2124.05002439]
 [  2365.65288887]
 [  2001.82331314]
 [  5050.11742108]
 [  1873.36625554]
 [  1873.36625554]
 [  1873.36625554]
 [  1873.36625554]
 [ 13033.93275207]
 [  2688.02791551]
 [  3862.24907869]
 [  2365.65288887]
 [  2336.39627142]
 [  2773.44160075]
 [  1873.36625554]
 [  1873.36625554]
 [  1526.43521463]
 [  1286.09171366]
 [  2034.09588696]
 [  7713.61507356]
 [  1873.36625554]
 [  3045.56960396]
 [  1873.36625554]
 [  1873.36625554]
 [  1873.36625554]
 [  1873.36625554]
 [  1032.22659007]
 [  6963.05424183]
 [  2619.51801246]
 [  1873.36625554]
 [  2709.47214988]
 [  3975.29906205]
 [  1873.36625554]
 [  2463.32883322]
 [  2828.68290475]
 [  2336.39627142]
 [  2799.4262873 ]
 [  5239.91840605]
 [  1661.02000851]
 [  2364.06560477]
 [  1873.36625554]
 [  3484.2580694 ]
 [  2000.29881734]
 [  3987.65714468]
 [  3387.98549501]
 [  1873.36625554]
 [  1873.36625554]
 [  7222.134

0.39671025891021217

from sklearn.linear_model import LogisticRegression
model = LogisticRegression(solver = 'lbfgs')
model.fit(train_img, train_lbl)
y_pred = model.predict(test_img)
# how did our model perform?
count_misclassified = (test_lbl != y_pred).sum()
print('Misclassified samples: {}'.format(count_misclassified))
accuracy = metrics.accuracy_score(test_lbl, y_pred)
print('Accuracy: {:.2f}'.format(accuracy))


Logistic Regression for multiclass

In [0]:
from sklearn.model_selection import LeaveOneOut ########LEAVE ONE OUT
import numpy as np
from sklearn.linear_model import LogisticRegression


model = LogisticRegression(solver = 'lbfgs')

#clf.fit(x_train,y_train_true)
#y_test_pred=clf.predict(x_test)
#x_train = data_avn_fan.iloc[:,1:17]
'''
x_train = data_op.iloc[:,1:8] #----------------
y_train = data_op.iloc[:,[0]]
'''
import numpy as np
data_avn_op = pd.read_csv('op_avn_label.csv', header = 0)
data_fan_op = pd.read_csv('op_fan_label.csv', header = 0)
x_train = data_avn_op.iloc[:,1:16] #----------------
y_train = data_avn_op.iloc[:,[0]]

x = np.array(x_train)
y = np.array(y_train.values.ravel())
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
x = scaler.fit_transform(x)

x2 = np.array(x_train)
y = np.array(y_train.values.ravel())
x = scaler.fit_transform(x2) # with or without same result

loo = LeaveOneOut()
loo.get_n_splits(x)
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        #print (y_train_true.values.ravel(),y_train_pred)

        #y_test_pred = model.predict(x_test)        
#model.fit(x_train, y_train_true.values.ravel()) ###########################################

from sklearn import metrics
from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
print (classification_report(y_train, y_train_pred) )

Train Accuracy: 0.656488549618
Confusion Matrix :
[[19 30]
 [15 67]]
Accuracy Score : 0.656488549618
Report : 
             precision    recall  f1-score   support

          1       0.56      0.39      0.46        49
          2       0.69      0.82      0.75        82

avg / total       0.64      0.66      0.64       131



Train Accuracy: 0.664122137405
Confusion Matrix :
[[26 23]
 [21 61]]
Accuracy Score : 0.664122137405
Report : 
             precision    recall  f1-score   support

          1       0.55      0.53      0.54        49
          2       0.73      0.74      0.73        82

avg / total       0.66      0.66      0.66       131

#avn_Op
Train Accuracy: 0.656488549618
Confusion Matrix :
[[19 30]
 [15 67]]
Accuracy Score : 0.656488549618
Report : 
             precision    recall  f1-score   support

          1       0.56      0.39      0.46        49
          2       0.69      0.82      0.75        82

avg / total       0.64      0.66      0.64       131

# logistic Regression
# avn,fan,op
Train Accuracy: 0.717557251908
Confusion Matrix :
[[26 23]
 [14 68]]
Accuracy Score : 0.717557251908
Report : 
             precision    recall  f1-score   support

          1       0.65      0.53      0.58        49
          2       0.75      0.83      0.79        82

avg / total       0.71      0.72      0.71       131
# avn_fan
Train Accuracy: 0.748091603053
Confusion Matrix :
[[29 20]
 [13 69]]
Accuracy Score : 0.748091603053
Report : 
             precision    recall  f1-score   support

          1       0.69      0.59      0.64        49
          2       0.78      0.84      0.81        82

avg / total       0.74      0.75      0.74       131
# avn
Train Accuracy: 0.664122137405
Confusion Matrix :
[[27 22]
 [22 60]]
Accuracy Score : 0.664122137405
Report : 
             precision    recall  f1-score   support

          1       0.55      0.55      0.55        49
          2       0.73      0.73      0.73        82

avg / total       0.66      0.66      0.66       131
# fan
Train Accuracy: 0.641221374046
Confusion Matrix :
[[ 4 45]
 [ 2 80]]
Accuracy Score : 0.641221374046
Report : 
             precision    recall  f1-score   support

          1       0.67      0.08      0.15        49
          2       0.64      0.98      0.77        82

avg / total       0.65      0.64      0.54       131
# op
Accuracy: 0.641221374046
Confusion Matrix :
[[ 2 47]
 [ 0 82]]
Accuracy Score : 0.641221374046
Report : 
             precision    recall  f1-score   support

          1       1.00      0.04      0.08        49
          2       0.64      1.00      0.78        82

avg / total       0.77      0.64      0.52       131

In [0]:
from sklearn.neural_network import MLPClassifier
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import LeaveOneOut ########LEAVE ONE OUT
import numpy as np
parameters = {'solver': ['lbfgs'], 'max_iter': [1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000 ], 'alpha': 10.0 ** -np.arange(1, 10), 'hidden_layer_sizes':np.arange(10, 15), 'random_state':[0,1,2,3,4,5,6,7,8,9]}
model = GridSearchCV(MLPClassifier(), parameters, n_jobs=-1)
'''
print('Best C:',model.best_estimator_.C) 
print('Best Kernel:',model.best_estimator_.kernel)
print('Best Gamma:',model.best_estimator_.gamma)
'''

# M L P neural network

In [0]:
from sklearn.neural_network import MLPClassifier
import numpy as np
from sklearn.model_selection import LeaveOneOut ########LEAVE ONE OUT

model = MLPClassifier(solver='lbfgs', alpha=1e-5,hidden_layer_sizes=(100,100,100))
 ##For small datasets, however, ‘lbfgs’ can converge faster and perform better.

x_train = data_avn_fan_op.iloc[:,1:25] #----------------
y_train = data_avn_fan_op.iloc[:,[0]]


x = np.array(x_train)
y = np.array(y_train.values.ravel())
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
x = scaler.fit_transform(x)

loo = LeaveOneOut()
loo.get_n_splits(x)
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        #print (y_train_true.values.ravel(),y_train_pred)

        #y_test_pred = model.predict(x_test)        
#model.fit(x_train, y_train_true.values.ravel()) ###########################################


from sklearn import metrics
from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
print (classification_report(y_train, y_train_pred) )

Train Accuracy: 0.916030534351
Confusion Matrix :
[[48  1]
 [10 72]]
Accuracy Score : 0.916030534351
Report : 
             precision    recall  f1-score   support

          1       0.83      0.98      0.90        49
          2       0.99      0.88      0.93        82

avg / total       0.93      0.92      0.92       131



# MLP solver='lbfgs', alpha=1e-5,hidden_layer_sizes=(100,100,100)
# avn fan op
Accuracy: 0.916030534351
Confusion Matrix :
[[48  1]
 [10 72]]
Accuracy Score : 0.916030534351
Report : 
             precision    recall  f1-score   support

          1       0.83      0.98      0.90        49
          2       0.99      0.88      0.93        82

avg / total       0.93      0.92      0.92       131
# fan op
Confusion Matrix :
[[44  5]
 [22 60]]
Accuracy Score : 0.793893129771
Report : 
             precision    recall  f1-score   support

          1       0.67      0.90      0.77        49
          2       0.92      0.73      0.82        82

avg / total       0.83      0.79      0.80       131
# avn op
Confusion Matrix :
[[39 10]
 [ 9 73]]
Accuracy Score : 0.854961832061
Report : 
             precision    recall  f1-score   support

          1       0.81      0.80      0.80        49
          2       0.88      0.89      0.88        82

avg / total       0.85      0.85      0.85       131
 
 # avn fan
Confusion Matrix :
[[46  3]
 [12 70]]
Accuracy Score : 0.885496183206
Report : 
             precision    recall  f1-score   support

          1       0.79      0.94      0.86        49
          2       0.96      0.85      0.90        82

avg / total       0.90      0.89      0.89       131 
# avn
Confusion Matrix :
[[39 10]
 [26 56]]
Accuracy Score : 0.725190839695
Report : 
             precision    recall  f1-score   support

          1       0.60      0.80      0.68        49
          2       0.85      0.68      0.76        82

avg / total       0.76      0.73      0.73       131
# fan
Confusion Matrix :
[[43  6]
 [25 57]]
Accuracy Score : 0.763358778626
Report : 
             precision    recall  f1-score   support

          1       0.63      0.88      0.74        49
          2       0.90      0.70      0.79        82

avg / total       0.80      0.76      0.77       131
# op
Confusion Matrix :
[[19 30]
 [ 2 80]]
Accuracy Score : 0.75572519084
Report : 
             precision    recall  f1-score   support

          1       0.90      0.39      0.54        49
          2       0.73      0.98      0.83        82

avg / total       0.79      0.76      0.72       131


In [0]:
#x_train = data_op.iloc[:,1:8] #----------------
#y_train = data_op.iloc[:,[0]]
import numpy as np
data_avn_op = pd.read_csv('op_avn_label.csv', header = 0)
data_fan_op = pd.read_csv('op_fan_label.csv', header = 0)
x_train = data_op.iloc[:,1:8] #----------------
y_train = data_op.iloc[:,[0]]
model = MLPClassifier(solver='lbfgs', alpha=1e-5,hidden_layer_sizes=(100,100,100))
'''
data_avn = 1:8

data_fan 1:8

data_avn_fan  1:17

data_avn_fan_op = 1:25

data_op 1:8
'''

'\ndata_avn = 1:8\n\ndata_fan 1:8\n\ndata_avn_fan  1:17\n\ndata_avn_fan_op = 1:25\n\ndata_op 1:8\n'

In [0]:
x = np.array(x_train)
y = np.array(y_train.values.ravel())
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
x = scaler.fit_transform(x)

loo = LeaveOneOut()
loo.get_n_splits(x)
# x_train mane whole data nea dorkar ekhane
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        
from sklearn import metrics
from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
y_train_pred =  model.predict(x_train)
print (classification_report(y_train, y_train_pred) )

Confusion Matrix :
[[19 30]
 [ 2 80]]
Accuracy Score : 0.75572519084
Report : 
             precision    recall  f1-score   support

          1       0.90      0.39      0.54        49
          2       0.73      0.98      0.83        82

avg / total       0.79      0.76      0.72       131



In [0]:
from sklearn import datasets
from sklearn import metrics
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt    
    
    plt.style.use('ggplot')

from xgboost import XGBClassifier
from catboost import CatBoostClassifier
from lightgbm import LGBMClassifier
    
    # load the wine datasets
dataset = datasets.load_wine()
X = dataset.data; y = dataset.target
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25)
    
    # fit a XGBoost model to the data
model = XGBClassifier()
model.fit(X_train, y_train)
print(); print(model)
    
    # make predictions
expected_y  = y_test
predicted_y = model.predict(X_test)

    # summarize the fit of the model
print(); print('XGBoost: ')
print(); print(metrics.classification_report(expected_y, predicted_y, 
                   target_names=dataset.target_names))
print(); print(metrics.confusion_matrix(expected_y, predicted_y))

    # fit a CatBoost model to the data
model = CatBoostClassifier()
model.fit(X_train, y_train)
    print(); print(model)

    # make predictions
expected_y  = y_test
predicted_y = model.predict(X_test)

    # summarize the fit of the model
print(); print('CatBoost: ')
print(); print(metrics.classification_report(expected_y, predicted_y, 
                   target_names=dataset.target_names))
print(); print(metrics.confusion_matrix(expected_y, predicted_y))
    
    # fit a LightGBM model to the data
model = LGBMClassifier()
model.fit(X_train, y_train)
print(); print(model)

    # make predictions
expected_y  = y_test
predicted_y = model.predict(X_test)
    
    # summarize the fit of the model
print(); print('LightGBM: ')
print(); print(metrics.classification_report(expected_y, predicted_y, 
                   target_names=dataset.target_names))
print(); print(metrics.confusion_matrix(expected_y, predicted_y))